### PR TITLE
Add create, list, delete volunteer use cases for DG

### PR DESCRIPTION
**Pull request by brein62**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -320,6 +320,75 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1.  User requests to list all volunteers
+2.  iVolunteer shows a list of all volunteers
+
+    Use case ends.
+
+**Use case: Delete all volunteers**
+
+**MSS**
+
+1.  User requests to list all volunteers
```
Comment (jordankanghm) [inline docs/DeveloperGuide.md:372]: I think this part can reference the list all volunteers use case?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -320,6 +320,75 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2.  iVolunteer shows a list of all volunteers
+3.  User requests to delete a specific volunteer in the volunteer list
+4.  iVolunteer removes the volunteer from all events he/she is in
+5.  iVolunteer deletes the volunteer
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
```
Comment (jordankanghm) [inline docs/DeveloperGuide.md:382]: maybe can have iVolunteer inform the user that there is no such volunteer?

**Review by jordankanghm — APPROVED** [review]
lgtm!!!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (brein62) [reply docs/DeveloperGuide.md:372]: for now i will name my one UCV01 - UCV03 until we come up with a standardised name for it

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (brein62) [reply docs/DeveloperGuide.md]: donded

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (brein62) [reply docs/DeveloperGuide.md]: done, rest can follow up for this

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -320,6 +320,83 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Use case UCV02: List all volunteers**
+
+**MSS**
+
+1.  User requests to list all volunteers.
+2.  iVolunteer shows a list of all volunteers.
+
+    Use case ends.
+
+**Use case UCV03: Delete all volunteers**
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:368]: I think you meant `delete a volunteer` ?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -320,6 +320,83 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case resumes from step 2.
+
+**Use case UCV04: Clear all volunteers in volunteer list**
+
+**MSS**
+
+1. User clears all volunteers in volunteer list.
+2. The volunteer list becomes empty.
+
+    Use case ends.
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:398]: Do we really need a feature to delete all volunteers? Feels like won't be necessary

**Review by zacwong2151 — CHANGES_REQUESTED** [review]
Need clarity for some parts

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (brein62) [reply docs/DeveloperGuide.md:368]: oops

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (brein62) [reply docs/DeveloperGuide.md:398]: this one is analogous to `clear` in AB3 - clears volunteers **from view** but doesn't really delete any volunteer

Comment (brein62) [reply docs/DeveloperGuide.md:398]: or else the `vlist` a bit no point HAHA

Comment (zacwong2151) [reply docs/DeveloperGuide.md:398]: oki

**Review by zacwong2151 — APPROVED** [review]

**Pull request merged**